File: App/Static/scripts/ts/classes/entities/Column.ts
import {Task} from "./Task";

export class Column
{
	private readonly _id: number;
	private _name: string;
	private readonly _tasks: Task[];
	
	constructor(id: number, name: string, tasks: Task[])
	{
		this._id = id;
		this._name = name;
		this._tasks = tasks;
	}
	
	get name(): string
	{
		return this._name;
	}
	
	set name(value: string)
	{
		this._name = value;
	}
	
	get tasks(): Task[]
	{
		return this._tasks;
	}
}

File: DataAccessLayer/Entities/Column.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using DataAccessLayer.Entities.Constraints.Abstract;

namespace DataAccessLayer.Entities
{
	public class Column
	{
		public Column()
		{
			Tasks       = new List<Task>();
			Constraints = new List<ColumnConstraintEntityBase>();
		}

		public int Id { get; set; }

		[Required]
		[MaxLength(255)]
		public string Name { get; set; }

		public virtual History History   { get; set; }
		public         int     HistoryId { get; set; }

		public virtual Board Board   { get; set; }
		public         int   BoardId { get; set; }

		public virtual User Creator   { get; set; }
		public         int  CreatorId { get; set; }

		public virtual ICollection<ColumnConstraintEntityBase> Constraints { get; set; }
		public virtual ICollection<Task>                       Tasks       { get; set; }
	}
}